lib/jekyll/reader.rb

```diff
@@ -19,6 +19,7 @@ def read
       @site.data = DataReader.new(site).read(site.config["data_dir"])
       CollectionReader.new(site).read
       ThemeAssetsReader.new(site).read
+      ThemeDataReader.new(site).read(site.config["data_dir"])
```

Comment:
Themes should definitely only read from `_data` so they're standardized (we do this for all other directories I believe). That way if a user customizes their data dir, then they don't break their usage of the theme. Let's do `read(site.theme&.data_path)` here. Ideally we just use `DataReader` here too.

Reply:
> I think we should add this feature. I'd love to be able to reduce the differences between the ThemeDataReader and the Jekyll::DataReader classes so we can reuse the logic in the latter.

Then the main concern of @ashmaroli wouldn’t be addressed, since the DataReader recursively reads everything within `_data`. But since this holds also true for `_layouts` or `_includes` from end user POV I suggest it’s the desired behaviour.

To use the `DataReader` instead of the clone the merging of the data must happen within the `Reader`. That would mean a new method should do the handling of reading and merging the data.

Does this sound okay?